src/OrchardCore/OrchardCore.Infrastructure/Security/ApiAuthorizationHandler.cs

```diff
@@ -0,0 +1,63 @@
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace OrchardCore.Security
+{
+    /// <summary>
+    /// Provides a delegating logic for API authentication.
+    /// If no specific scheme handler is found it returns an anonymous user.
+    /// </summary>
+    public class ApiAuthorizationHandler : AuthenticationHandler<ApiAuthorizationOptions>
+    {
+        private readonly IOptions<AuthenticationOptions> _authenticationOptions;
+        private readonly IAuthenticationService _authenticationService;
+        private static readonly AuthenticateResult Anonymous;
+
+        static ApiAuthorizationHandler()
+        {
+            Anonymous = AuthenticateResult.Success(
```

Comment:
Nah, this is evil. Having a "fake" (or more exactly delegating here) handler is a thing, but using fake authenticated identities is another and IMHO, we shouldn't go that far.

Using `AuthenticateResult.None()` or better `AuthenticateResult.Fail()` with an error message indicating there's no handler available to process tokens is the right thing to do.